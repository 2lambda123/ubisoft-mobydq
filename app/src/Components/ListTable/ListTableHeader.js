import React from 'react';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import TableCell from '@material-ui/core/TableCell';
import TableSortLabel from '@material-ui/core/TableSortLabel';
import Tooltip from '@material-ui/core/Tooltip';

/**
 * Component to build the list table header.
 * @param headerNames array of column names
 */
class ListTableHeader extends React.Component {
  _handleSortClick = (fieldName, isActive, direction) => {
    const newDirection = (isActive && direction === SORT_ORDER.DESCENDING) ? 'ASC' : 'DESC';
    this.props.params.setSortColumn(fieldName + '_' + newDirection);
  };

  /**
  * GraphQL uses FIELD_NAME format for defining the sort directives as opposed to camelCase in the field names.
  * This function is responsible for the necessary field name conversion
  **/
  _buildSortableHeaderCell(header, sortHeader, sortActive, sortDirection) {
    return header === 'Actions'
      ? <React.Fragment>{header}</React.Fragment>
      : <Tooltip
          title='Sort'
          placement={'bottom-start'}
          enterDelay={300}
        >
          <TableSortLabel
            active={sortActive}
            direction={sortDirection}
            onClick={() => this._handleSortClick(sortHeader, sortActive, sortDirection)}
          >
            {header}
          </TableSortLabel>
        </Tooltip>;
  }

  _buildHeaderCell(fieldName, sortField, sortDirection) {
    let header = fieldName;
    let sortHeader = '';
   if (header.length > 0) {
      header = header.charAt(0).toUpperCase() + header.slice(1);
      sortHeader = header.match(/[A-Z][a-z]+/ug).join('_').toUpperCase();
      header = header.match(/[A-Z][a-z]+/ug).join(' ');
    }
    sortDirection = (sortField && sortHeader === sortField) ? sortDirection.toLowerCase() : SORT_ORDER.DESCENDING;
    const sortActive = sortField && sortHeader === sortField && fieldName !== 'Actions';
    return (
      <TableCell
        key={header}
      >
        {this._buildSortableHeaderCell(header, sortHeader, sortActive, sortDirection)}
      </TableCell>
    );
  }

  _buildHeader(headerFields) {
    // Add "Actions column
    const sortField = this.props.params.sortColumn;
    const allHeaderFields = headerFields.concat(['Actions']);
    //sort field is either null or FIELD_NAME_ASC/DESC
    const sortFieldName = sortField === null ? null : sortField.substring(0, sortField.lastIndexOf("_"));
    const sortDirection = sortField === null ? null : sortField.substring(sortField.lastIndexOf("_") + 1);
    return (
      <TableHead>
        <TableRow>
          {allHeaderFields.map((fieldName) => this._buildHeaderCell(fieldName, sortFieldName, sortDirection))}
        </TableRow>
      </TableHead>
    );
  }

  render() {
    return this._buildHeader(this.props.headerNames);
  }
}

const SORT_ORDER = {
    'DESCENDING': 'desc',
    'ASCENDING': 'asc'
}

export default ListTableHeader;


