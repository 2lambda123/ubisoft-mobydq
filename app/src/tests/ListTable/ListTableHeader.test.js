import React from 'react'
import * as Adapter from 'enzyme-adapter-react-16';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import ListTableHeader from './../../Components/ListTable/ListTableHeader'

describe('ListTableHeader unit test', () => {
  let wrapper;
  let headerNames = ['id', 'name', 'longName', ''];

  beforeEach(() => {
    wrapper = shallow(<ListTableHeader headerNames={headerNames}/>)
  });

  it('renders', () => {
    expect(wrapper.length).toEqual(1);
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('ListTableHeader functional test', () => {
  it ('renders properly capitalizes the names and renders the action column', () => {
    let headerNames = ['id', 'name', 'longName', ''];
    let wrapper = mount(<table><ListTableHeader headerNames={headerNames}/></table>);
    expect(wrapper.find('tr').length).toEqual(1);
    expect(wrapper.find('th').length).toEqual(5);
    expect(wrapper.find('th').at(0).text()).toEqual('Id');
    expect(wrapper.find('th').at(1).text()).toEqual('Name');
    expect(wrapper.find('th').at(2).text()).toEqual('Long Name');
    expect(wrapper.find('th').at(3).text()).toEqual('');
    expect(wrapper.find('th').at(4).text()).toEqual('Actions');
  });
  it ('renders correctly with no headers', () => {
    let headerNames = [];
    let wrapper = mount(<table><ListTableHeader headerNames={headerNames}/></table>);
    expect(wrapper.find('tr').length).toEqual(1);
    expect(wrapper.find('th').length).toEqual(1);
    expect(wrapper.find('th').at(0).text()).toEqual('Actions');

  });
});