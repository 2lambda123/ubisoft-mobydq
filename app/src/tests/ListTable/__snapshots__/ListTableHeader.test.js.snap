// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListTableHeader unit test matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ListTableHeader
    headerNames={
      Array [
        "id",
        "name",
        "longName",
        "",
      ]
    }
    params={
      Object {
        "sortColumn": null,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <WithStyles(TableRow)>
        <WithStyles(TableCell)>
          <WithStyles(Tooltip)
            enterDelay={300}
            placement="bottom-start"
            title="Sort"
          >
            <WithStyles(TableSortLabel)
              active={null}
              direction="desc"
              onClick={[Function]}
            >
              Id
            </WithStyles(TableSortLabel)>
          </WithStyles(Tooltip)>
        </WithStyles(TableCell)>
        <WithStyles(TableCell)>
          <WithStyles(Tooltip)
            enterDelay={300}
            placement="bottom-start"
            title="Sort"
          >
            <WithStyles(TableSortLabel)
              active={null}
              direction="desc"
              onClick={[Function]}
            >
              Name
            </WithStyles(TableSortLabel)>
          </WithStyles(Tooltip)>
        </WithStyles(TableCell)>
        <WithStyles(TableCell)>
          <WithStyles(Tooltip)
            enterDelay={300}
            placement="bottom-start"
            title="Sort"
          >
            <WithStyles(TableSortLabel)
              active={null}
              direction="desc"
              onClick={[Function]}
            >
              Long Name
            </WithStyles(TableSortLabel)>
          </WithStyles(Tooltip)>
        </WithStyles(TableCell)>
        <WithStyles(TableCell)>
          <WithStyles(Tooltip)
            enterDelay={300}
            placement="bottom-start"
            title="Sort"
          >
            <WithStyles(TableSortLabel)
              active={null}
              direction="desc"
              onClick={[Function]}
            >
              
            </WithStyles(TableSortLabel)>
          </WithStyles(Tooltip)>
        </WithStyles(TableCell)>
        <WithStyles(TableCell)>
          <React.Fragment>
            Actions
          </React.Fragment>
        </WithStyles(TableCell)>
      </WithStyles(TableRow)>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(TableCell)>
            <WithStyles(Tooltip)
              enterDelay={300}
              placement="bottom-start"
              title="Sort"
            >
              <WithStyles(TableSortLabel)
                active={null}
                direction="desc"
                onClick={[Function]}
              >
                Id
              </WithStyles(TableSortLabel)>
            </WithStyles(Tooltip)>
          </WithStyles(TableCell)>,
          <WithStyles(TableCell)>
            <WithStyles(Tooltip)
              enterDelay={300}
              placement="bottom-start"
              title="Sort"
            >
              <WithStyles(TableSortLabel)
                active={null}
                direction="desc"
                onClick={[Function]}
              >
                Name
              </WithStyles(TableSortLabel)>
            </WithStyles(Tooltip)>
          </WithStyles(TableCell)>,
          <WithStyles(TableCell)>
            <WithStyles(Tooltip)
              enterDelay={300}
              placement="bottom-start"
              title="Sort"
            >
              <WithStyles(TableSortLabel)
                active={null}
                direction="desc"
                onClick={[Function]}
              >
                Long Name
              </WithStyles(TableSortLabel)>
            </WithStyles(Tooltip)>
          </WithStyles(TableCell)>,
          <WithStyles(TableCell)>
            <WithStyles(Tooltip)
              enterDelay={300}
              placement="bottom-start"
              title="Sort"
            >
              <WithStyles(TableSortLabel)
                active={null}
                direction="desc"
                onClick={[Function]}
              >
                
              </WithStyles(TableSortLabel)>
            </WithStyles(Tooltip)>
          </WithStyles(TableCell)>,
          <WithStyles(TableCell)>
            <React.Fragment>
              Actions
            </React.Fragment>
          </WithStyles(TableCell)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "Id",
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Tooltip)
              enterDelay={300}
              placement="bottom-start"
              title="Sort"
            >
              <WithStyles(TableSortLabel)
                active={null}
                direction="desc"
                onClick={[Function]}
              >
                Id
              </WithStyles(TableSortLabel)>
            </WithStyles(Tooltip)>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(TableSortLabel)
                active={null}
                direction="desc"
                onClick={[Function]}
              >
                Id
              </WithStyles(TableSortLabel)>,
              "enterDelay": 300,
              "placement": "bottom-start",
              "title": "Sort",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "active": null,
                "children": "Id",
                "direction": "desc",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Id",
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "Name",
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Tooltip)
              enterDelay={300}
              placement="bottom-start"
              title="Sort"
            >
              <WithStyles(TableSortLabel)
                active={null}
                direction="desc"
                onClick={[Function]}
              >
                Name
              </WithStyles(TableSortLabel)>
            </WithStyles(Tooltip)>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(TableSortLabel)
                active={null}
                direction="desc"
                onClick={[Function]}
              >
                Name
              </WithStyles(TableSortLabel)>,
              "enterDelay": 300,
              "placement": "bottom-start",
              "title": "Sort",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "active": null,
                "children": "Name",
                "direction": "desc",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Name",
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "Long Name",
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Tooltip)
              enterDelay={300}
              placement="bottom-start"
              title="Sort"
            >
              <WithStyles(TableSortLabel)
                active={null}
                direction="desc"
                onClick={[Function]}
              >
                Long Name
              </WithStyles(TableSortLabel)>
            </WithStyles(Tooltip)>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(TableSortLabel)
                active={null}
                direction="desc"
                onClick={[Function]}
              >
                Long Name
              </WithStyles(TableSortLabel)>,
              "enterDelay": 300,
              "placement": "bottom-start",
              "title": "Sort",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "active": null,
                "children": "Long Name",
                "direction": "desc",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Long Name",
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "",
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Tooltip)
              enterDelay={300}
              placement="bottom-start"
              title="Sort"
            >
              <WithStyles(TableSortLabel)
                active={null}
                direction="desc"
                onClick={[Function]}
              >
                
              </WithStyles(TableSortLabel)>
            </WithStyles(Tooltip)>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(TableSortLabel)
                active={null}
                direction="desc"
                onClick={[Function]}
              >
                
              </WithStyles(TableSortLabel)>,
              "enterDelay": 300,
              "placement": "bottom-start",
              "title": "Sort",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "active": null,
                "children": "",
                "direction": "desc",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "",
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "Actions",
          "nodeType": "class",
          "props": Object {
            "children": <React.Fragment>
              Actions
            </React.Fragment>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Actions",
            },
            "ref": null,
            "rendered": "Actions",
            "type": Symbol(react.fragment),
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(TableRow)>
          <WithStyles(TableCell)>
            <WithStyles(Tooltip)
              enterDelay={300}
              placement="bottom-start"
              title="Sort"
            >
              <WithStyles(TableSortLabel)
                active={null}
                direction="desc"
                onClick={[Function]}
              >
                Id
              </WithStyles(TableSortLabel)>
            </WithStyles(Tooltip)>
          </WithStyles(TableCell)>
          <WithStyles(TableCell)>
            <WithStyles(Tooltip)
              enterDelay={300}
              placement="bottom-start"
              title="Sort"
            >
              <WithStyles(TableSortLabel)
                active={null}
                direction="desc"
                onClick={[Function]}
              >
                Name
              </WithStyles(TableSortLabel)>
            </WithStyles(Tooltip)>
          </WithStyles(TableCell)>
          <WithStyles(TableCell)>
            <WithStyles(Tooltip)
              enterDelay={300}
              placement="bottom-start"
              title="Sort"
            >
              <WithStyles(TableSortLabel)
                active={null}
                direction="desc"
                onClick={[Function]}
              >
                Long Name
              </WithStyles(TableSortLabel)>
            </WithStyles(Tooltip)>
          </WithStyles(TableCell)>
          <WithStyles(TableCell)>
            <WithStyles(Tooltip)
              enterDelay={300}
              placement="bottom-start"
              title="Sort"
            >
              <WithStyles(TableSortLabel)
                active={null}
                direction="desc"
                onClick={[Function]}
              >
                
              </WithStyles(TableSortLabel)>
            </WithStyles(Tooltip)>
          </WithStyles(TableCell)>
          <WithStyles(TableCell)>
            <React.Fragment>
              Actions
            </React.Fragment>
          </WithStyles(TableCell)>
        </WithStyles(TableRow)>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(TableCell)>
              <WithStyles(Tooltip)
                enterDelay={300}
                placement="bottom-start"
                title="Sort"
              >
                <WithStyles(TableSortLabel)
                  active={null}
                  direction="desc"
                  onClick={[Function]}
                >
                  Id
                </WithStyles(TableSortLabel)>
              </WithStyles(Tooltip)>
            </WithStyles(TableCell)>,
            <WithStyles(TableCell)>
              <WithStyles(Tooltip)
                enterDelay={300}
                placement="bottom-start"
                title="Sort"
              >
                <WithStyles(TableSortLabel)
                  active={null}
                  direction="desc"
                  onClick={[Function]}
                >
                  Name
                </WithStyles(TableSortLabel)>
              </WithStyles(Tooltip)>
            </WithStyles(TableCell)>,
            <WithStyles(TableCell)>
              <WithStyles(Tooltip)
                enterDelay={300}
                placement="bottom-start"
                title="Sort"
              >
                <WithStyles(TableSortLabel)
                  active={null}
                  direction="desc"
                  onClick={[Function]}
                >
                  Long Name
                </WithStyles(TableSortLabel)>
              </WithStyles(Tooltip)>
            </WithStyles(TableCell)>,
            <WithStyles(TableCell)>
              <WithStyles(Tooltip)
                enterDelay={300}
                placement="bottom-start"
                title="Sort"
              >
                <WithStyles(TableSortLabel)
                  active={null}
                  direction="desc"
                  onClick={[Function]}
                >
                  
                </WithStyles(TableSortLabel)>
              </WithStyles(Tooltip)>
            </WithStyles(TableCell)>,
            <WithStyles(TableCell)>
              <React.Fragment>
                Actions
              </React.Fragment>
            </WithStyles(TableCell)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "Id",
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(Tooltip)
                enterDelay={300}
                placement="bottom-start"
                title="Sort"
              >
                <WithStyles(TableSortLabel)
                  active={null}
                  direction="desc"
                  onClick={[Function]}
                >
                  Id
                </WithStyles(TableSortLabel)>
              </WithStyles(Tooltip)>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <WithStyles(TableSortLabel)
                  active={null}
                  direction="desc"
                  onClick={[Function]}
                >
                  Id
                </WithStyles(TableSortLabel)>,
                "enterDelay": 300,
                "placement": "bottom-start",
                "title": "Sort",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "active": null,
                  "children": "Id",
                  "direction": "desc",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Id",
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "Name",
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(Tooltip)
                enterDelay={300}
                placement="bottom-start"
                title="Sort"
              >
                <WithStyles(TableSortLabel)
                  active={null}
                  direction="desc"
                  onClick={[Function]}
                >
                  Name
                </WithStyles(TableSortLabel)>
              </WithStyles(Tooltip)>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <WithStyles(TableSortLabel)
                  active={null}
                  direction="desc"
                  onClick={[Function]}
                >
                  Name
                </WithStyles(TableSortLabel)>,
                "enterDelay": 300,
                "placement": "bottom-start",
                "title": "Sort",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "active": null,
                  "children": "Name",
                  "direction": "desc",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Name",
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "Long Name",
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(Tooltip)
                enterDelay={300}
                placement="bottom-start"
                title="Sort"
              >
                <WithStyles(TableSortLabel)
                  active={null}
                  direction="desc"
                  onClick={[Function]}
                >
                  Long Name
                </WithStyles(TableSortLabel)>
              </WithStyles(Tooltip)>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <WithStyles(TableSortLabel)
                  active={null}
                  direction="desc"
                  onClick={[Function]}
                >
                  Long Name
                </WithStyles(TableSortLabel)>,
                "enterDelay": 300,
                "placement": "bottom-start",
                "title": "Sort",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "active": null,
                  "children": "Long Name",
                  "direction": "desc",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Long Name",
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "",
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(Tooltip)
                enterDelay={300}
                placement="bottom-start"
                title="Sort"
              >
                <WithStyles(TableSortLabel)
                  active={null}
                  direction="desc"
                  onClick={[Function]}
                >
                  
                </WithStyles(TableSortLabel)>
              </WithStyles(Tooltip)>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <WithStyles(TableSortLabel)
                  active={null}
                  direction="desc"
                  onClick={[Function]}
                >
                  
                </WithStyles(TableSortLabel)>,
                "enterDelay": 300,
                "placement": "bottom-start",
                "title": "Sort",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "active": null,
                  "children": "",
                  "direction": "desc",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "",
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "Actions",
            "nodeType": "class",
            "props": Object {
              "children": <React.Fragment>
                Actions
              </React.Fragment>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Actions",
              },
              "ref": null,
              "rendered": "Actions",
              "type": Symbol(react.fragment),
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
